smart_open/tests/test_gcs.py

```diff
@@ -172,6 +170,7 @@ def exists(self, client=None):
         return self._exists
 
     def upload_from_string(self, str_):
+        self.__contents = io.BytesIO()
```

Comment:
You can be more terse:

```suggestion
        self.__contents = io.BytesIO(str_)
```

Also, the method name is misleading. You're talking about strings, but you're actually writing bytes. If this is part of Google's API, then we can't really do much, other than leave a comment here to console people working on this in the future.

Reply:
This is part of Google's API, the method names are misleading on their end. I added comments and improved variable names to document this